internal/querynode/task.go

```diff
@@ -377,15 +377,22 @@ func (w *watchDmChannelsTask) Execute(ctx context.Context) error {
 	)
 
 	// add flow graph
-	channel2FlowGraph := w.node.dataSyncService.addFlowGraphsForDMLChannels(collectionID, vChannels)
+	channel2FlowGraph, err := w.node.dataSyncService.addFlowGraphsForDMLChannels(collectionID, vChannels)
+	if err != nil {
+		log.Warn("watchDMChannel, add flowGraph for dmChannels failed", zap.Int64("collectionID", collectionID), zap.Strings("vChannels", vChannels), zap.Error(err))
```

Comment:
when err != nil, the returned channel2FlowGraph is nil. This process of cleaning up the Flow graph is invalid